trunk/SUAVE/Components/Energy/Storages/Batteries/Constant_Mass/Lithium_Ion_LiFePO4_18650.py

```diff
@@ -0,0 +1,362 @@
+        battery.state_of_charge                    = SOC_new 
+        battery.depth_of_discharge                 = DOD_new
+        battery.charge_throughput                  = Q_total 
+        battery.cell_charge_throughput             = Q_total/n_parallel  
+        battery.voltage_open_circuit               = V_oc
+        battery.voltage_under_load                 = V_ul
+        battery.current                            = I_bat 
+        battery.pack_temperature                   = T_current 
+        battery.cell_joule_heat_fraction           = np.zeros_like(V_ul)
+        battery.cell_entropy_heat_fraction         = np.zeros_like(V_ul)  
+        battery.cell_voltage_open_circuit          = np.zeros_like(V_ul)
+        battery.cell_current                       = np.zeros_like(V_ul)
+        battery.thevenin_voltage                   = np.zeros_like(V_ul)
+        battery.heat_energy_generated              = Q_heat_gen 
+        battery.internal_resistance                = R_0 
+        battery.cell_voltage_under_load            = V_ul 
+        
+        return      
+    
+    def append_battery_unknowns(self,segment): 
+        """ Appends unknowns specific to LFP cells which are unpacked from the mission solver and send to the network.
+    
+            Assumptions:
+            None
+    
+            Source:
+            N/A
+    
+            Inputs:
+            state.unknowns.battery_voltage_under_load               [volts]
+    
+            Outputs: 
+            state.conditions.propulsion.battery_voltage_under_load  [volts]
+    
+            Properties Used:
+            N/A
+        """             
+        
+        segment.state.conditions.propulsion.battery_voltage_under_load  = segment.state.unknowns.battery_voltage_under_load
+        
+        return 
+    
+    def append_battery_residuals(self,segment,network): 
+        """ Packs the residuals specific to LFP cells to be sent to the mission solver.
+    
+            Assumptions:
+            None
+    
+            Source:
+            N/A
+    
+            Inputs:
+            state.conditions.propulsion:
+                motor_torque                          [N-m]
+                propeller_torque                      [N-m]
+                voltage_under_load                    [volts]
+            state.unknowns.battery_voltage_under_load [volts]
+            
+            Outputs:
+            None
+    
+            Properties Used:
+            network.voltage                           [volts]
+        """     
+        v_actual  = segment.state.conditions.propulsion.battery_voltage_under_load
+        v_predict = segment.state.unknowns.battery_voltage_under_load
+        v_max     = network.voltage
+        
+        # Return the residuals
+        segment.state.residuals.network.voltage = (v_predict[:,0] - v_actual[:,0])/v_max
+        
+        return 
+    
+    def append_battery_unknowns_and_residuals_to_segment(self,segment,initial_voltage, 
```

Comment:
same comment; function already exists in base class

Reply:
Addressed